sections/sprint-4/updated-uppaal.tex

```diff
@@ -0,0 +1,144 @@
+In this section the \uppTemp{Host} uses the \uppOut{TCP_send} synchronisation to notify each \uppTemp{Client} that the game is starting through TCP.
+Finally the \uppTemp{Host} is in-game and continuously sends messages to the \uppTemp{Clients}.
+This is done through either \uppOut{TCP_send} synchronisation or \uppOut{UDP_send} synchronisation which are the TCP and UDP channels respectively.
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=1\linewidth]{sprint4/Host.png}
+    \caption{The Host template of the \uppaal model.}
+    \label{fig:sprint4-uppaal-host}
+\end{figure}
+\begin{uppaalcode}[caption={local Host declarations}, label={lst:sprint-4-host-code}]
+
+int connected_clients = 0;
+int local_TCP_buffer;
+int i = 0;
+\end{uppaalcode}
+
+
+% Client
+\subsection{Client template}
+Like the \uppTemp{Host} template, the \uppTemp{Client} has a local declaration for saving the data it reads from the global buffer.
+Since it is desirable to have a variable amount of instances, the template has an \uppType{id_t} parameter called \uppVar{id}, to allow instantiating multiple client processes.
+The \uppTemp{Client} has four primary locations, as seen on \autoref{fig:sprint4-uppaal-client}, which correspond to the states that the actual implementation can be in.
+These four can be split into two groups\\
+The \uppLoc{Initial} location which is the state of the client before it attempts to connect to a host, \uppLoc{Connecting} for when it is connecting to a hosted game instance, and the \uppLoc{Lobby} connection for when the connection has been established, but the game has not yet started.
+Finally, there is the \uppLoc{InGame} location, which is where the client will be located most of the time during the simulation. 
+This indicates that the client is now in a game, and can continuously receive information about the game state from the host either via UDP or TCP.
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=1\linewidth]{sprint4/Client.png}
+    \caption{The Client template of the \uppaal model.}
+    \label{fig:sprint4-uppaal-client}
+\end{figure}
+
+% TCP
+\subsection{TCP host and client templates}
+The \uppTemp{TCP_Host} and \uppTemp{TCP_Client} were designed such that there will be a process of each for each client process that exists.
+This was designed to model the socket connection that is established between the host and a client.
+Like in the \uppTemp{Client}, this happens by having an \uppVar{clientId} as a parameter for when the process is instantiated.
+As seen on \autoref{fig:sprint4-uppaal-tcp-client} and \autoref{fig:sprint4-uppaal-tcp-host}, the TCP templates will stay in a \uppLoc{Start} location until they receive an \uppSync{input syncronization} from either a \uppTemp{Host} or \uppTemp{Client}.
+The \uppTemp{TCP_Host} and \uppTemp{TCP_Client} will then begin to initialize a 3-way handshake to ensure that both parties are ready for transferring data between them.
```

Comment:
hvis vi skriver synchronisation skal vi vel skrive initialise også for at være true brit 🤔 ?